docker-compose.base.yml

```diff
@@ -1,22 +1,37 @@
 services:
+    minio:
+        image: minio/minio:latest
+        healthcheck:
+            test: ["CMD", "curl", "-f", "http://localhost:9000/minio/health/ready"]
+            interval: 1m30s
+            timeout: 30s
+            retries: 5
+            start_period: 30s
+        restart: unless-stopped
+        command: server --console-address ":9001" /data
+        ports:
+            - "9000:9000" # API
+            - "9001:9001" # Console
+
+    minio-init:
+        image: minio/mc:latest
+        networks:
+            - app-network
+        entrypoint: >
+            /bin/sh -c '
+            /usr/bin/mc config host add myminio http://minio:9000 "$${MINIO_ROOT_USER}" "$${MINIO_ROOT_PASSWORD}";
+            /usr/bin/mc mb --ignore-existing myminio/"$${ASSETS_BUCKET_NAME}";
+            /usr/bin/mc anonymous set download myminio/"$${ASSETS_BUCKET_NAME}";
+            /usr/bin/mc admin user add myminio "$${ASSETS_UPLOAD_KEY}" "$${ASSETS_UPLOAD_SECRET_KEY}";
+            /usr/bin/mc admin policy attach myminio readwrite --user "$${ASSETS_UPLOAD_KEY}";
+            exit 0;
+            '
```

Comment:
it might be better to just have this in a script.

basically what this does:
- create the bucket
- allow public downloads for said bucket
- add a new user
- allow said user to upload

this sadly can't be done by just configuring some env vars for minio itself, very annoying, you need to use their `mc` client. I'm not sure this is the best way to go about doing this.

Reply:
Could you add this explanation as a comment to the yml file?